src/store/index.js

```diff
@@ -0,0 +1,109 @@
+import { createStore } from "vuex";
+import axios from "axios";
+import router from "../router";
+
+const store = createStore({
+  state: {
+    posts: [],
+    showDeleteAlert: false,
+    post: {
+      image: "",
+      tags: [
+        "transport",
+        "sports",
+        "people",
+        "nightlife",
+        "nature",
+        "food",
+        "fashion",
+        "city",
+        "cats",
+        "business",
+        "other",
+      ],
+      selectedTags: [],
+      description: "",
+    },
+  },
+  getters: {
+    sortedPosts: (state) => {
+      return state.posts
+        .slice()
+        .sort((a, b) => new Date(b.created_at) - new Date(a.created_at));
+    },
+  },
+  mutations: {
+    SET_POSTS(state, posts) {
+      state.posts = posts;
+    },
+    REMOVE_DELETED_POST(state, deletedPostId) {
+      state.posts = state.posts.filter((post) => post.id !== deletedPostId);
+    },
+    SET_SHOW_CREATE_ALERT(state, value) {
+      state.showCreateAlert = value;
+    },
+    SET_SHOW_DELETE_ALERT(state, value) {
+      state.showDeleteAlert = value;
+    },
+  },
+  actions: {
+    fetchPosts({ commit }) {
+      return axios
+        .get("http://localhost:3000/posts")
```

Comment:
Extract your backend's base URL into a constant variable. Ideally, this should be an environment variable.
You can create a store variable or put it above the store. i.e.
`const BACKEND_URL = http://localhost:3000`

You can now access everywhere in this file e.g. `.get(`${BACKEND_URL}/posts") `. We will see how to configure this in Axios directly in the future.